src/payment/views.py

```diff
@@ -0,0 +1,139 @@
+
+
+class SubscriptionViewSet(viewsets.ModelViewSet):
+    stripe.api_key = settings.STRIPE_API_KEY
+    stripe.api_version = settings.STRIPE_API_VERSION
+    plan_name = settings.ESSENTIAL_PLAN_NAME
+
+    queryset = Subscription.objects.all()
+    serializer_class = SubscriptionSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(user=self.request.user)
+
+    def create(self, request, *args, **kwargs):
+        user = request.user
+        plan = request.data.get("plan")  # "monthly" ou "annual"
+
+        try:
+            product = StripeProduct.objects.get(name=self.plan_name)
+            price_id = (
+                product.monthly_price_id
+                if plan == "monthly"
+                else product.annual_price_id
+            )
+        except StripeProduct.DoesNotExist:
+            return Response({"error": "Product not found"}, status=404)
+
+        try:
+            checkout_session = stripe.checkout.Session.create(
+                payment_method_types=["card"],
+                customer_email=user.email,
+                line_items=[{"price": price_id, "quantity": 1}],
+                mode="subscription",
+                billing_address_collection="required",
+                currency="eur",
+                success_url=request.build_absolute_uri(
+                    reverse("v1:payment:subscription-success")
+                ),
+                cancel_url=request.build_absolute_uri(
+                    reverse("v1:payment:subscription-cancel")
+                ),
+                metadata={
+                    "user_id": user.id,
+                    "product_name": product.name,
+                },
+            )
+
+            data = {
+                **request.data,
+                "stripe_subscription_id": checkout_session.id,
+            }
+
+            headers = self.get_success_headers(data)
+            return Response(
+                {
+                    "sessionId": checkout_session.id,
+                    "checkout_url": checkout_session.url,
+                },
+                status=status.HTTP_201_CREATED,
+                headers=headers,
+            )
+        except Exception as e:
+            print("Error during session creation:", e)
+            return Response({"error": str(e)}, status=500)
+
+    def retrieve(self, request, *args, **kwargs):
+        obj = get_object_or_404(self.get_queryset(), user=request.user)
+        serializer = self.get_serializer(obj)
+        return Response(serializer.data)
+
+
+class SubscriptionSuccessView(APIView):
+    def get(self, request, *args, **kwargs):
+        return Response(
+            {"message": "Subscription successful"}, status=status.HTTP_200_OK
+        )
+
+
+class SubscriptionCancelView(APIView):
+    def get(self, request, *args, **kwargs):
+        return Response(
+            {"message": "Subscription cancelled"}, status=status.HTTP_200_OK
+        )
```

Comment:
Actually, this is the correct code because Stripe requires this specific functionality to confirm subscription cancellations. The endpoint should return a success message with a 200 OK status. If we were to return a 501 status code, it would indicate that the feature is not implemented, which isn't the case here since cancellation is handled.

Reply:
So it's a stripe callback? Let's play with this later to make sure the workflow is clear for both of us

Reply:
This endpoint corresponds to the URL Stripe redirects users to after a session if the payment fails or is canceled. Specifically, when creating a checkout session, we define a cancel_url where Stripe redirects the user if the payment is not successful. This ensures that the user lands on a proper cancellation page, confirming the cancellation action with a 200 OK status.

Reply:
Yeah that makes sense, let's just add the docstring below SubscriptionCancelView

Reply:
I've added the docstrings to both SubscriptionSuccessView and SubscriptionCancelView to clarify their purpose and integration within the Stripe workflow